pkg/ottl/ottlfuncs/README.md

```diff
@@ -587,6 +588,40 @@ Examples:
 
 - `ExtractPatterns(body, "^(?P<timestamp>\\w+ \\w+ [0-9]+:[0-9]+:[0-9]+) (?P<hostname>([A-Za-z0-9-_]+)) (?P<process>\\w+)(\\[(?P<pid>\\d+)\\])?: (?P<message>.*)$")`
 
+### ExtractGrokPatterns
+
+`ExtractGrokPatterns(target, pattern, Optional[namedCapturesOnly], Optional[patternDefinitions])`
+
+The `ExtractGrokPatterns` Converter returns a `pcommon.Map` struct that is a result of extracting named capture groups from the target string. If no matches are found then an empty `pcommon.Map` is returned.
+
+`target` is a Getter that returns a string. `pattern` is a grok pattern string. `namedCapturesOnly` specifies if non-named captures should be returned. `patternDefinitions` is a list of custom pattern definition strings used `pattern` in a form of `PATTERN_NAME=PATTERN`. 
+
+Use `patternDefinition` to improve readability when extracted `pattern` is not part of the default set or you need custom naming. 
+For example to parse password from `/etc/passwd` and keep `pattern` readable:
+- `pattern`: `%{USERNAME:user.name}:%{PASSWORD:user.password}:%{USERINFO}`
+- `patternDefinitions` as `USERNAME` is in default set:
+  - `PASSWORD=%{WORD}`
+  - `USERINFO=%{GREEDYDATA}`
+- `smith:x:1001:1000:J Smith,1234,(234)567-8910,(234)567-1098,email:/home/smith:/bin/sh` resulting in:
+ - `user.name`: smith
+ - `user.password`: pass123
```

Comment:
I'm still pretty confused about what patternDefinitions is doing.  Where did the value `pass123` come from? I don't see it in the example string.

Reply:
it's second part of the string, my bad there was x previously

Reply:
So what is 

```
  - `PASSWORD=%{WORD}`
  - `USERINFO=%{GREEDYDATA}
```

Providing in this scenario? It looks to me like `%{USERNAME:user.name}:%{PASSWORD:user.password}` is doing all the work

Reply:
`%{USERNAME:user.name}:%{PASSWORD:user.password}` is a pattern so it is a format in which your logs occur in a file. 

```
  - `PASSWORD=%{WORD}`
  - `USERINFO=%{GREEDYDATA}
```

is a pattern definition, it tells parser how `%{USERNAME:user.name}` translates to reges, in this case it is a `WORD` that comes from a default set and is `\b\w+\b`([definition](https://github.com/elastic/go-grok/blob/84fd21b31b32a4aa6848dd8c2ad8365826f53162/patterns/default.go#L21C15-L21C23))

so you after compilation you end up with named capture <user.name> capturing regex `\b\w+\b`

Reply:
I asked very nice people from our docs team to help me rephrase this so it is more clear